src/main/java/com/alex/eshop/eshop/Service/CategoryServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package com.alex.eshop.eshop.Service;
+
+import com.alex.eshop.eshop.Entity.CategoryEntity;
+import com.alex.eshop.eshop.Repository.CategoryRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Component
+@Transactional
+public class CategoryServiceImpl implements CategoryService{
+    private final CategoryRepository categoryRepository;
+    @Override
+    public List<CategoryEntity> getAllCategories() {
+        return categoryRepository.findAll();
+    }
+    public CategoryEntity getCategory(int id){
```

Comment:
`CategoryEntity getCategory(int id)` should not return NULL. if category does not exists.
Return category, or throw exception, if category does not exists.
Show informative message in exception. Also return 404 http status code in response.
Use `.orElseThrow();` for Optional